roles/docker/tasks/systemcontainer_docker.yml

```diff
@@ -0,0 +1,106 @@
+---
+# If docker_options are provided we should fail. We should not install docker and ignore
+# the users configuration. NOTE: docker_options == inventory:openshift_docker_options
+- name: Fail quickly if openshift_docker_options are set
+  assert:
+    that:
+      - docker_options is defined
+      - docker_options != ""
+    msg: |
+      Docker via System Container does not allow for the use of the openshift_docker_options
+      variable. If you want to use openshift_docker_options you will need to use the
+      traditional docker package install. Otherwise, comment out openshift_docker_options
+      in your inventory file.
+
+# Used to pull and install the system container
+- name: Ensure atomic is installed
+  package:
+    name: atomic
+    state: present
+
+# At the time of writing the atomic command requires runc for it's own use. This
+# task is here in the even that the atomic package ever removes the dependency.
+- name: Ensure runc is installed
+  package:
+    name: runc
+    state: present
+
+# If we are on atomic, set http_proxy and https_proxy in /etc/atomic.conf
+- name: Set http(s) proxies on atomic host
+  block:
+
+    - name: Add http_proxy to /etc/atomic.conf
+      lineinfile:
+        path: /etc/atomic.conf
+        line: "http_proxy={{ openshift.common.http_proxy | default('') }}"
+      when:
+        - openshift.common.http_proxy is defined
+        - openshift.common.http_proxy != ''
```

Comment:
If you set the proxy with the installer, remove the proxy and run the installer again it will add the proxy line.  If you set the proxy manually then run the installer it will have 2 proxy lines and could be problematic. However, I don't believe there should be a case where an admin goes and manually configured the proxy and then expects the installer to to remove it and replace it with it's own.

Though I do agree, I'd like a more solid solution later on if possible.